### PR TITLE
fetch all feature flags at the same time

### PR DESCRIPTION
**Pull request by AMarcedone**
cc @mlsteele for the journeycard change.

### CODE REVIEW

**Comment by mlsteele** [comment]
Oh I get it. Yes, this is better thanks.

**Comment by heronhaye** [comment]
@mlsteele why is it better from the journeycardmanager's perspective?

**Comment by mlsteele** [comment]
It is better because in the second-saddest case master does this:
1. Start an http request to get the state
2. Time's up
3. Cancel the http request
4. Start another http request in the background to store the state for later

But with Amarctx it does this:
1. Start an http request to get the state
2. Time's up
3. The http request continues and when completed stores the state for later

**Comment by AMarcedone** [comment]
@mlsteele with this pr all features are now syncing together: whenever you update one they all do in the same rpc call. 
Maybe we can add a background job to keep them all up to to date (so they are ready for use when needed), but not sure that is worth it right now...

File: go/chat/journey_card_manager.go [hunk 1]
```diff
@@ -188,27 +188,34 @@ func NewJourneyCardManagerSingleUser(g *globals.Context, ri func() chat1.RemoteI
...
-	enabled, err := cc.G().FeatureFlags.EnabledWithError(cc.MetaContext(ctx), libkb.FeatureJourneycard)
-	if err != nil {
-		// Send one out in a goroutine to get the goods for next time.
-		ctx, cancel2 := context.WithTimeout(globals.BackgroundChatCtx(ogCtx, cc.G()), 10*time.Second)
-		defer cancel2()
-		go cc.G().FeatureFlags.Enabled(cc.MetaContext(ctx), libkb.FeatureJourneycard)
+	var err error
+	doneChan := make(chan struct{})
+	go func() {
+		enabled, err = cc.G().FeatureFlags.EnabledWithError(cc.MetaContext(ctx), libkb.FeatureJourneycard)
```
[the remark this answers is not in the document]
Comment (AMarcedone) [reply go/chat/journey_card_manager.go]: I was going off of this https://golang.org/ref/spec#Receive_operator (reading a closed channel is always possible in a select statement), but what you suggest is probably more readable

[the remark this answers is not in the document]
Comment (heronhaye) [reply go/chat/journey_card_manager.go]: Maybe @mlsteele could grace us with his opinion

[the remark this answers is not in the document]
Comment (mlsteele) [reply go/chat/journey_card_manager.go]: lol whatever works

[the remark this answers is not in the document]
Comment (AMarcedone) [reply go/chat/journey_card_manager.go]: Updated.

**Pull request merged**